src/agent/testresultreader.ts

```diff
@@ -0,0 +1,337 @@
+import ifm = require('./api/interfaces');
+import trp = require('./testrunpublisher');
+
+var fs = require('fs');
+var path = require("path");
+var xmlreader = require('xmlreader');
+
+export class JUnitResultReader implements trp.IResultReader {
+    public readResults(file: string, runContext: trp.TestRunContext) {
+        
+        var testRun2 : ifm.TestRun2;
+        var contents = fs.readFileSync(file, "utf-8");
+        contents = contents.replace("\ufeff", ""); //replace BOM if exits to avoid xml read error
+      
+        var buildId = runContext.buildId;
+        var buildRequestedFor = runContext.requestedFor;
+        var platform = "";
+        var config = "";
+
+        xmlreader.read(contents, function (err, res){
```

Comment:
Since this is an async method, the function would return before this gets a change to complete.  Per above, I think this method needs a callback and the callback needs to pass a callback